### PR TITLE
ChildBuilder docs improvement

### LINKED ISSUE #7230: Improve `ChildBuilder` documentation to ensure a consistent style
## How can Bevy's documentation be improved?

The documentation added in #6035 is not great (see the suggestions at the bottom of the thread), but I decided to merge this relatively optimistically to help unblock an old PR.

We should clean up that documentation, and do a pass on all of the methods in that trait for consistency and clarity.

### PR DESCRIPTION
**Pull request by JohnTheCoolingFan**
# Objective

Fixes #7230 

## Solution

Documented some side-effects

### CODE REVIEW

File: crates/bevy_hierarchy/src/child_builder.rs
Comment (nicopap) [inline crates/bevy_hierarchy/src/child_builder.rs:116]: I think the following might improve comprehension, to me `children` here is not explicit enough.

Additionally, splitting in two sentences helps clarity.

```suggestion
/// Removes entities in `children` from `parent`'s [`Children`], removing the component if ends up empty.
/// Also removes [`Parent`] components from entities in `children`.
```

File: crates/bevy_hierarchy/src/child_builder.rs
Comment (nicopap) [inline crates/bevy_hierarchy/src/child_builder.rs:162]: :+1: I love when sentences are terminated with a dot.

File: crates/bevy_hierarchy/src/child_builder.rs
Comment (nicopap) [inline crates/bevy_hierarchy/src/child_builder.rs:249]: :+1:

File: crates/bevy_hierarchy/src/child_builder.rs
Comment (nicopap) [inline crates/bevy_hierarchy/src/child_builder.rs:19]:
```suggestion
/// Adds `child` to `parent`'s [`Children`], eithout checking if it is already present there.
```

File: crates/bevy_hierarchy/src/child_builder.rs
Comment (nicopap) [inline crates/bevy_hierarchy/src/child_builder.rs:257]: This sentence is not English. This needs a revision. I think the following might help: Usually `ChildBuilder` is a variable named `parent`. Consider that `self` _is_ `parent`.

File: crates/bevy_hierarchy/src/child_builder.rs
Comment (nicopap) [inline crates/bevy_hierarchy/src/child_builder.rs:266]: Same remark: `self` _is_ `parent`, this need rephrasing to account for it.

File: crates/bevy_hierarchy/src/child_builder.rs
Comment (nicopap) [inline crates/bevy_hierarchy/src/child_builder.rs:316]:
```suggestion
    /// The removed children will have their [`Parent`] component removed.
```

File: crates/bevy_hierarchy/src/child_builder.rs
Comment (nicopap) [inline crates/bevy_hierarchy/src/child_builder.rs:322]: :+1:

File: crates/bevy_hierarchy/src/child_builder.rs
Comment (nicopap) [inline crates/bevy_hierarchy/src/child_builder.rs:278]: I think this should remove references to `ChildBuilder` since it's irrelevant to what the added command does. Here is an imperfect suggestion:

```suggestion
    /// Adds a command, like [`Commands::add`].
```

**Review by nicopap — CHANGES_REQUESTED** [review]
I think this PR is value added, but need some improvements still.

File: crates/bevy_hierarchy/src/child_builder.rs
Comment (nicopap) [inline crates/bevy_hierarchy/src/child_builder.rs:19]: What are the implications of adding `child` without checking? Could you add a line explaining why you should really not do it?

File: crates/bevy_hierarchy/src/child_builder.rs
Comment (nicopap) [inline crates/bevy_hierarchy/src/child_builder.rs:285]:
```suggestion
/// Trait for removing, adding and replacing children and parents of an entity.
```

File: crates/bevy_hierarchy/src/child_builder.rs
Comment (nicopap) [inline crates/bevy_hierarchy/src/child_builder.rs:278]:
```suggestion
    /// Adds a command to be executed, like [`Commands::add`].
```

File: crates/bevy_hierarchy/src/child_builder.rs
Comment (nicopap) [inline crates/bevy_hierarchy/src/child_builder.rs:225]:
```suggestion
/// Command that clear all children from an entity, replacing them with the given children.
```

**Review by nicopap — CHANGES_REQUESTED** [review]
Some minor changes and I think it's good to go.

File: crates/bevy_hierarchy/src/child_builder.rs
Comment (JohnTheCoolingFan) [reply crates/bevy_hierarchy/src/child_builder.rs:19]: All I know about unchecked push is that might lead to the same entity appearing in the list twice. I don't know what problems that might cause. Mostly documented just to tell what's *not* checked, since function name `push_child_unchecked` doesn't really tell what checks it skips.

Comment (nicopap) [reply crates/bevy_hierarchy/src/child_builder.rs:19]: It's probably going to result in unexpected results when removing duplicate children.

The PR needs a second approval anyway, so can we keep this unresolved to get a second opinion?

```suggestion
/// Adds `child` to `parent`'s [`Children`], without checking if it is already present there.
///
/// This might cause unexpected results when removing duplicate children.
```

**Review by nicopap — APPROVED** [review]
I'm still wondering if the wording about "this entity's parent" in `ChildBuilder` is not misleading. But it's already a net improvement, so it's good for me.

**Review by alice-i-cecile — APPROVED** [review]
A substantial improvement. Approving and merging.

**Pull request merged**
